### ISSUE TITLE
bin/logstash-plugin update doesnt work

### ISSUE THREAD

**Issue by ph** [elastic/logstash#6339]
Steps to reproduce with an artefact.

1. install a previous version of a plugin
``
bin/logstash-plugin install --version 0.1.0 logstash-filter-qatest
``

2. Edit Gemfile and remove the version constrains.

3. try one of this command

```
bin/logstash-plugin update logstash-filter-qatest
bin/logstash-plugin update
```

You will get the following output.

```
ph@sashimi~/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT  master ± bin/logstash-plugin update --no-verify logstash-filter-qatest
Bundler::GemNotFound: Could not find logstash-filter-qatest-0.1.1 in any of the sources
      materialize at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/spec_set.rb:92
             map! at org/jruby/RubyArray.java:2446
      materialize at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/spec_set.rb:85
            specs at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/definition.rb:132
        specs_for at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/definition.rb:177
  requested_specs at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/definition.rb:166
  requested_specs at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/environment.rb:18
            setup at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler/runtime.rb:13
            setup at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/bundler-1.9.10/lib/bundler.rb:122
           setup! at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/lib/bootstrap/bundler.rb:64
           (root) at /Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/lib/pluginmanager/main.rb:9
```

OR
```
No plugin  updated
```

**Comment by ph** [comment]
I think my lock file is not consistant I will regenerate the package and test again.

**Comment by ph** [comment]
Ok, I know mostly what is happening the problem is with when we call `LogStash::bundler::invoke(:clean => true)`  at https://github.com/elastic/logstash/blob/master/lib/pluginmanager/update.rb#L65

For the xpack install I had to change where and when the current installed specs where loaded and I believe that this has created this side effect.

If you add `DEBUG=1` before running the update or the install command, theses steps are happening in that order

1. Successfully resolve dependencies
2. The last version of the qatest filter get installed
3. bundler.invoke(:clean  => true) remove the qatest filter because it cannot find it in the memory spec.
4. The lock was updated on disk with the last qatest
5. The plugin cannot be found because the folder was removed during step 3.


I think for now we should just remove the clean step when we run the update command, it shouldn't not interfere with logstash since the unused gems won be added to the `$LOAD_PATH`
